### PR TITLE
FEATURE: Use Contact Picker API for invites

### PR DESCRIPTION
**Pull request by xfalcox**
output

### CODE REVIEW

File: app/assets/javascripts/discourse/components/invite-panel.js.es6 [hunk 1]
```diff
@@ -180,6 +182,10 @@ export default Component.extend({
     );
   },
 
+  showContactPicker: Ember.computed(function() {
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/components/invite-panel.js.es6]: would be better to use `computed` directly here, didn't do it when I gave you this example.

```
import { computed } from "@ember/object";

showContactPicker: computed(function() {
  return this.capabilities.hasContactPicker;
});
```

File: app/assets/javascripts/discourse/components/invite-panel.js.es6 [hunk 2]
```diff
@@ -433,6 +439,13 @@ export default Component.extend({
       } else {
         this.set("customMessage", null);
       }
+    },
+
+    searchContact() {
+      getNativeContact(['email'], false)
+        .then(result => {
+          this.setProperties({emailOrUsername: result[0].email[0]});
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/components/invite-panel.js.es6]:
```suggestion
          this.set("emailOrUsername", result[0].email[0]);
```

File: app/assets/javascripts/discourse/lib/pwa-utils.js.es6 [hunk 3]
```diff
@@ -26,3 +26,20 @@ export function nativeShare(data) {
...
+  return new Promise((resolve, reject) => {
+    if (!caps.hasContactPicker) {
+      reject();
+      return;
+    }
+
+    navigator.contacts
+      .select(properties, {multiple: multiple})
+      .then(resolve)
+      .catch(() => {
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/lib/pwa-utils.js.es6]:
```suggestion
      .catch(reject)
```

Comment (jjaffeux) [inline app/assets/javascripts/discourse/lib/pwa-utils.js.es6]:
```
 .catch(reject)
```

File: app/assets/javascripts/discourse/lib/pwa-utils.js.es6 [hunk 4]
```diff
@@ -26,3 +26,20 @@ export function nativeShare(data) {
...
+export function getNativeContact(properties, multiple) {
+  const caps = Discourse.__container__.lookup("capabilities:main");
+  return new Promise((resolve, reject) => {
+    if (!caps.hasContactPicker) {
+      reject();
+      return;
+    }
+
+    navigator.contacts
+      .select(properties, {multiple: multiple})
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/lib/pwa-utils.js.es6]:
```suggestion
      .select(properties, { multiple })
```

**Comment by xfalcox** [comment]
> This looks quite cool but the button is definitely the wrong size there. Can it be customized?

Moved it. Also asked @awesomerobot  to take a look last week, as this UI using input group is not something we do in Discourse, but I couldn't come with a better one.

**Pull request merged**